nio/client/async_client.py

```diff
@@ -146,6 +166,8 @@ def __init__(
 
         self.sharing_session = dict()  # type: Dict[str, Event]
 
+        config = config or AsyncClientConfig()
```

Comment:
It would be nice for compatibility reasons and to make the API easier to use if the client would convert a `ClientConfig` into an `AsyncClientConfig` here. That way users can pick the wrong config and it won't crash their code.